### PR TITLE
Include kubernetes-client config in all-config page

### LINKED ISSUE #16052: Quarkus "all config" guide is missing kubernetes-client config
## Describe the bug
The Quarkus All Config guide (https://quarkus.io/guides/all-config) doesn't seem to have any of the `kubernetes-client` config.

I haven't looked at others so can't speak if it's missing other config as well.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #16052

### CODE REVIEW

**Comment by geoand** [comment]
I know that these changes are needed to make the docs work, but I wasn't able to verify that the kubernetes client config actually ends up in the generated docs because I was not able to verify the same for configuration that already does exist in that page (for example `quarkus.container-image.*` items).

@gsmet any pointers here would be appreciated :)

**Comment by aloubyansky** [comment]
Did have the wrong artifactId all over the place?

**Comment by geoand** [comment]
The problem with the artifact name (and the reason I changed it) was that it was causing a NPE in the `GeneratePlatformDescriptorJsonMojo`.
So I changed all instances of that deployment artifact name

**Review by aloubyansky — APPROVED** [review]

File: extensions/kubernetes-client/deployment-internal/pom.xml [hunk 1]
```diff
@@ -9,8 +9,8 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <artifactId>quarkus-kubernetes-client-deployment-internal</artifactId>
-    <name>Quarkus - Kubernetes Client - Deployment - Internal</name>
+    <artifactId>quarkus-kubernetes-client-internal-deployment</artifactId>
+    <name>Quarkus - Kubernetes Client - Internal- Deployment</name>
```
Comment (gsmet) [inline extensions/kubernetes-client/deployment-internal/pom.xml:13]:
```suggestion
    <name>Quarkus - Kubernetes Client - Internal - Deployment</name>
```

**Review by gsmet — DISMISSED** [review]
Spotted a small typo.

I don't have the time to check what's happening with the docs at the moment. Maybe @machi1990 could have a look? If not, I'll have a look later this week.

**Comment by geoand** [comment]
> I don't have the time to check what's happening with the docs at the moment. Maybe @machi1990 could have a look? If not, I'll have a look later this week.

I will merge this in the the meantime as it's necessary anyway

**Comment by machi1990** [comment]
> I know that these changes are needed to make the docs work, but I wasn't able to verify that the kubernetes client config actually ends up in the generated docs because I was not able to verify the same for configuration that already does exist in that page (for example `quarkus.container-image.*` items).
> 
> @gsmet any pointers here would be appreciated :)

I can do the verification if this is still needed @geoand .

**Comment by geoand** [comment]
> I can do the verification if this is still needed @geoand .

Thanks. Once I merge it, feel free to check :)

**Comment by machi1990** [comment]
> > I can do the verification if this is still needed @geoand .
> 
> Thanks. Once I merge it, feel free to check :)

LGTM, I compiled the docs with `mvn clean install -DskipTests -DskipITs` and then ` cat target/asciidoc/generated/config/quarkus-all-config.adoc | grep "kubernetes-client"` showed entries like 
```
a|icon:lock[title=Fixed at build time] [[quarkus-kubernetes-client-internal_quarkus.kubernetes-client.trust-certs]]`link:#quarkus-kubernetes-client-internal_quarkus.kubernetes-client.trust-certs[quarkus.kubernetes-client.trust-certs]`
a|icon:lock[title=Fixed at build time] [[quarkus-kubernetes-client-internal_quarkus.kubernetes-client.master-url]]`link:#quarkus-kubernetes-client-internal_quarkus.kubernetes-client.master-url[quarkus.kubernetes-client.master-url]`
a|icon:lock[title=Fixed at build time] [[quarkus-kubernetes-client-internal_quarkus.kubernetes-client.namespace]]`link:#quarkus-kubernetes-client-internal_quarkus.kubernetes-client.namespace[quarkus.kubernetes-client.namespace]`

```

**Review by machi1990 — APPROVED** [review]

**Comment by geoand** [comment]
Thanks!

**Pull request merged**